fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/PermissiveAvailability.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.metadata.DataSourceMetadataService;
+import com.yahoo.bard.webservice.table.Column;
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint;
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * An availability which allows missing intervals, i.e. returns union of available intervals, on its contents.
+ */
+public class PermissiveAvailability extends ConcreteAvailability {
+
+    /**
+     * Constructor.
+     *
+     * @param tableName The name of the data source
+     * @param columns A set of columns associated with the data source
+     * @param metadataService A service containing the data source segment data
+     */
+    public PermissiveAvailability(
+            @NotNull TableName tableName,
+            @NotNull Set<Column> columns,
+            @NotNull DataSourceMetadataService metadataService
+    ) {
+        super(tableName, columns, metadataService);
+    }
+
+    /**
+     * Returns union of all available intervals.
+     * <p>
+     * This is different from its parent's
+     * {@link
+     * com.yahoo.bard.webservice.table.availability.ConcreteAvailability#getAvailableIntervals(DataSourceConstraint)};
+     * Instead of returning the intersection of all available intervals, this method returns the union of them.
+     *
+     * @param constraints  Data constraint containing columns and api filters
+     *
+     * @return the union of all available intervals
+     */
+    @Override
+    public SimplifiedIntervalList getAvailableIntervals(DataSourceConstraint constraints) {
+        return getIntervalsForColumnNames(getColumns().stream().map(Column::getName).collect(Collectors.toSet()))
```

Comment:
Rather than recompute this, we should pull this collection from the parent class: https://github.com/yahoo/fili/pull/190/files#diff-6923730d44e29354ccfcc8d1ca622d8eR33

We may need to add a `protected` getter for it to the parent, but that's where we should use it from.